app/src/main/java/ch/onepass/onepass/ui/organizerprofile/OrganizerProfileViewModel.kt

```diff
@@ -0,0 +1,232 @@
+import ch.onepass.onepass.model.event.EventRepository
+import ch.onepass.onepass.model.event.EventRepositoryFirebase
+import ch.onepass.onepass.model.organization.Organization
+import ch.onepass.onepass.model.organization.OrganizationRepository
+import ch.onepass.onepass.model.organization.OrganizationRepositoryFirebase
+import com.google.firebase.Timestamp
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asSharedFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+
+// --- UI Models ---
+
+/** Represents the selected tab in the organizer profile */
+enum class OrganizerProfileTab {
+  POSTS,
+  UPCOMING,
+  PAST
+}
+
+/** UI state for the organizer profile screen */
+data class OrganizerProfileUiState(
+    val organizationId: String = "",
+    val name: String = "",
+    val description: String = "",
+    val profileImageUrl: String? = null,
+    val websiteUrl: String? = null,
+    val instagramUrl: String? = null,
+    val tiktokUrl: String? = null,
+    val facebookUrl: String? = null,
+    val followersCount: Int = 0,
+    val isFollowing: Boolean = false,
+    val isVerified: Boolean = false,
+    val eventCount: Int = 0,
+    val selectedTab: OrganizerProfileTab = OrganizerProfileTab.UPCOMING,
+    val upcomingEvents: List<Event> = emptyList(),
+    val pastEvents: List<Event> = emptyList(),
+    val loading: Boolean = true,
+    val errorMessage: String? = null
+) {
+  /** Formatted followers count for display (e.g., "1K", "2.5M") */
+  val followersCountFormatted: String
+    get() = formatFollowersCount(followersCount)
+}
+
+/**
+ * Formats follower count for display:
+ * - < 1000: shows exact number (e.g., "500")
+ * - >= 1000 and < 1 million: shows with K suffix (e.g., "1K", "2.5K")
+ * - >= 1 million: shows with M suffix (e.g., "1M", "2.5M")
+ */
+private fun formatFollowersCount(count: Int): String {
+  return when {
+    count < 1000 -> count.toString()
+    count < 1_000_000 -> {
+      val thousands = count / 1000.0
+      if (thousands % 1 == 0.0) "${thousands.toInt()}K"
+      else {
+        // This fixes the rounding error where 999.95K would round to 1000.0K :)
+        val capped = thousands.coerceAtMost(999.9)
+        "%.1fK".format(capped)
+      }
+    }
+    else -> {
+      val millions = count / 1_000_000.0
+      if (millions % 1 == 0.0) "${millions.toInt()}M" else "%.1fM".format(millions)
+    }
+  }
+}
+
+/** One-time effects/events for navigation and side effects */
+sealed interface OrganizerProfileEffect {
+  data class NavigateToEvent(val eventId: String) : OrganizerProfileEffect
+
+  data class OpenWebsite(val url: String) : OrganizerProfileEffect
+
+  data class OpenSocialMedia(val platform: String, val url: String) : OrganizerProfileEffect
+
+  data class ShowError(val message: String) : OrganizerProfileEffect
+}
```

Comment:
Good clean sealed interface for one-time effects! But OpenSocialMedia takes platform string. 
What if caller passes invalid platform? Consider:

```suggestion
/** One-time effects/events for navigation and side effects */
sealed interface OrganizerProfileEffect {
    data class NavigateToEvent(val eventId: String) : OrganizerProfileEffect
    data class OpenWebsite(val url: String) : OrganizerProfileEffect
    
    sealed class OpenSocialMedia(val url: String) : OrganizerProfileEffect {
        data class Instagram(override val url: String) : OpenSocialMedia(url)
        data class TikTok(override val url: String) : OpenSocialMedia(url)
        data class Facebook(override val url: String) : OpenSocialMedia(url)
    }
    
    data class ShowError(val message: String) : OrganizerProfileEffect
}
```

Type-safe and self-documenting.